ci/compile_requirements.sh

```diff
@@ -1,13 +1,32 @@
 #!/bin/bash
-set -xeo pipefail
+set -xueo pipefail
+
+if [ -z ${PIP_COMPILE_ARGS+x} ]; then
+    PIP_COMPILE_ARGS=""
+fi
+
+if [[ "${ARROW_NIGHTLY+1}" == 1 ]]; then
+    ARROW_URL="$(./ci/get_pyarrow_nightly.py)"
+    ARROW_FILE="$(basename $ARROW_URL)"
+    ARROW_VERSION="$(echo "$ARROW_FILE" | awk '{split($1,a,"-"); print a[2]}')"
```

Comment:
you are nesting double quotes here

Reply:
I am nesting `"..."`, `$(...)`, `"..."`. That's valid in bash.

Reply:
We were talking about an arrow issue connected to packaging of nighties, weren't we? Can you attach a link to that ticket. I don't mind the messiness for now but I'd like to keep track no this. This doesn't feel like a stable solution, especially not in comparison to pandas/numpy

Reply:
my bad